### PR TITLE
rounded rectangle

### PR DESCRIPTION
**Pull request by aymey**
related to #43 

> auto switching between rounded and non-rounded

how would we do this? seems polymorphic so rust traits would be the only option right?

### CODE REVIEW

**Comment by aymey** [comment]
or a simpler solution would be a condition in `shape()` i.e.
```rust
impl VelloVector for VelloRect {
    fn shape(&self) -> impl kurbo::Shape {
       if self.radius == 0 { kurbo::Rect::new(...)}
       else { kurbo::RoundedRect::new(...)}
    }
}

```

**Comment by nixon-voxell** [comment]
yeah, I was thinking maybe just a simple if statement? haha not sure if that would worth the performance cost tho?

**Comment by aymey** [comment]
im currently struggling with object safety by return different types. Seems like a new `Shape` trait with object safety is necessary although it may lack features required for the objects manipulation.

As for performance, i **think** it would actually be much worse if we use return different types due to dynamic dispatch rather than if we just return a `RoundedRect` with a radius of 0 some of the time.

**Comment by nixon-voxell** [comment]
Oh yes, of course, I have not thought of that! haha thanks for pointing it out, then I guess, the easiest solution would be the current implementation or add a new type `VelloRoundedRect` tho, I think the current implementation is more user friendly.

**Comment by aymey** [comment]
I vote to keep the current implementation until https://github.com/linebender/kurbo/pull/331 gets merged then rewrite with that in mind

File: crates/bevy_vello_graphics/src/rect.rs [hunk 1]
```diff
@@ -9,13 +9,15 @@ use super::VelloVector;
 pub struct VelloRect {
     pub size: DVec2,
     pub anchor: DVec2,
+    pub radius: f64
```
Comment (nixon-voxell) [inline crates/bevy_vello_graphics/src/rect.rs:12]:
```suggestion
    pub radius: f64,
```

**Review by nixon-voxell — CHANGES_REQUESTED** [review]
Hey juz a tiny change to fix the rust fmt error, thanks!

**Review by nixon-voxell — APPROVED** [review]

**Comment by nixon-voxell** [comment]
@aymey I also wanted to point out that your changes will be moved into the new repo I created: https://github.com/voxell-tech/bevy_vello_graphics.

**Pull request merged**